akka-http2-support/src/main/scala/akka/http/impl/engine/http2/hpack/HeaderCompression.scala

```diff
@@ -55,6 +61,15 @@ private[http2] object HeaderCompression extends GraphStage[FlowShape[FrameEvent,
               }
             })
           }
+        case e: SyntheticFrameEvent ⇒
+          e match {
+            case SyntheticHpackEncoderSettingFrame(Setting(SETTINGS_HEADER_TABLE_SIZE, value)) ⇒
+              log.warning("Set outgoing compression table size to {}", value)
+              encoder.setMaxHeaderTableSize(os, value)
+              push(eventsOut, SettingsAckFrame)
```

Comment:
This wouldn't work if the setting came in together with another setting and both of them need to be ack'd together. => Let's just put the settings into the SettingsAck frame ;)